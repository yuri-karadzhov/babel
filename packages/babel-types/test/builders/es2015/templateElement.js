import * as t from "../../..";

describe("builders", function() {
  describe("es2015", function() {
    describe("templateElement", function() {
      it("should validate", function() {
        expect(
          t.templateElement({ raw: "foo", cooked: "foo" }),
        ).toMatchSnapshot();

        expect(t.templateElement({ raw: "foo" })).toMatchSnapshot();

        expect(() =>
          t.templateElement({ raw: 1 }),
        ).toThrowErrorMatchingSnapshot();

        expect(() =>
          t.templateElement({ raw: "foo", cooked: 1 }),
        ).toThrowErrorMatchingSnapshot();

        expect(() => t.templateElement("foo")).toThrowErrorMatchingSnapshot();
      });
    });
  });
});
