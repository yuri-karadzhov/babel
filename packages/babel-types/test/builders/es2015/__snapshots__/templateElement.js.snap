// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builders es2015 templateElement should validate 1`] = `
Object {
  "tail": false,
  "type": "TemplateElement",
  "value": Object {
    "cooked": "foo",
    "raw": "foo",
  },
}
`;

exports[`builders es2015 templateElement should validate 2`] = `
Object {
  "tail": false,
  "type": "TemplateElement",
  "value": Object {
    "raw": "foo",
  },
}
`;

exports[`builders es2015 templateElement should validate 3`] = `
"Property value of TemplateElement expected to have the following:
Property raw expected type of string but got number"
`;

exports[`builders es2015 templateElement should validate 4`] = `
"Property value of TemplateElement expected to have the following:
Property cooked expected type of string but got number"
`;

exports[`builders es2015 templateElement should validate 5`] = `
"Property value of TemplateElement expected to have the following:
Property raw expected type of string but got undefined"
`;
